tests/vkrenderframework.cpp

```diff
@@ -0,0 +1,2186 @@
+                desired_message_strings_.erase(desired_msg_it);
+                break;
+            }
+        }
+
+        if (!found_expected && allowed_message_strings_.size()) {
+            for (auto allowed_msg_it = allowed_message_strings_.begin(); allowed_msg_it != allowed_message_strings_.end();
+                 ++allowed_msg_it) {
+                if (errorString.find(*allowed_msg_it) != string::npos) {
+                    found_expected = true;
+                    break;
+                }
+            }
+        }
+
+        if (!found_expected) {
+            printf("Unexpected: %s\n", msgString);
+            other_messages_.push_back(errorString);
+        }
+    }
+
+    test_platform_thread_unlock_mutex(&mutex_);
+    return result;
+}
+
+vector<string> ErrorMonitor::GetOtherFailureMsgs() const { return other_messages_; }
+
+VkDebugReportFlagsEXT ErrorMonitor::GetMessageFlags() { return message_flags_; }
+
+bool ErrorMonitor::AnyDesiredMsgFound() const { return message_found_; }
+
+bool ErrorMonitor::AllDesiredMsgsFound() const { return desired_message_strings_.empty(); }
+
+void ErrorMonitor::SetError(const char *const errorString) {
+    test_platform_thread_lock_mutex(&mutex_);
+    message_found_ = true;
+    failure_message_strings_.insert(errorString);
+    test_platform_thread_unlock_mutex(&mutex_);
+}
+
+void ErrorMonitor::SetBailout(bool *bailout) {
+    test_platform_thread_lock_mutex(&mutex_);
+    bailout_ = bailout;
+    test_platform_thread_unlock_mutex(&mutex_);
+}
+
+void ErrorMonitor::DumpFailureMsgs() const {
+    vector<string> otherMsgs = GetOtherFailureMsgs();
+    if (otherMsgs.size()) {
+        std::cout << "Other error messages logged for this test were:" << std::endl;
+        for (auto iter = otherMsgs.begin(); iter != otherMsgs.end(); iter++) {
+            std::cout << "     " << *iter << std::endl;
+        }
+    }
+}
+
+void ErrorMonitor::ExpectSuccess(VkDebugReportFlagsEXT const message_flag_mask) {
+    // Match ANY message matching specified type
+    test_platform_thread_lock_mutex(&mutex_);
+    desired_message_strings_.insert("");
+    message_flags_ = message_flag_mask;
+    test_platform_thread_unlock_mutex(&mutex_);
+}
+
+void ErrorMonitor::VerifyFound() {
+    test_platform_thread_lock_mutex(&mutex_);
+    // Not receiving expected message(s) is a failure. /Before/ throwing, dump any other messages
+    if (!AllDesiredMsgsFound()) {
+        DumpFailureMsgs();
+        for (const auto &desired_msg : desired_message_strings_) {
+            ADD_FAILURE() << "Did not receive expected error '" << desired_msg << "'";
+        }
+    } else if (GetOtherFailureMsgs().size() > 0) {
+        // Fail test case for any unexpected errors
+#if defined(ANDROID)
+        // This will get unexpected errors into the adb log
+        for (auto msg : other_messages_) {
+            __android_log_print(ANDROID_LOG_INFO, "VulkanLayerValidationTests", "[ UNEXPECTED_ERR ] '%s'", msg.c_str());
```

Comment:
rename `VulkanLayerValidationTests`

if you search for `VulkanLayerValidationTests` you will find more